protos/proto/aptos/indexer/v1/grpc.proto

```diff
@@ -0,0 +1,108 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package aptos.indexer.v1;
+
+import "aptos/indexer/v1/raw_data.proto";
+import "aptos/transaction/v1/transaction.proto";
+import "aptos/util/timestamp/timestamp.proto";
+
+message StreamProgressSampleProto {
+  optional aptos.util.timestamp.Timestamp timestamp = 1;
+  uint64 version = 2;
+  uint64 size_bytes = 3;
+}
+
+message StreamProgress {
+  repeated StreamProgressSampleProto samples = 1;
+}
+
+message ActiveStream {
+  optional string id = 1;
+  optional aptos.util.timestamp.Timestamp start_time = 2;
+  uint64 start_version = 3;
+  optional uint64 end_version = 4;
+
+  optional StreamProgress progress = 5;
+}
+
+message StreamInfo {
+  repeated ActiveStream active_streams = 1;
+}
+
+message LiveDataServiceInfo {
+  optional aptos.util.timestamp.Timestamp timestamp = 1;
+  optional uint64 known_latest_version = 2;
+  optional StreamInfo stream_info = 3;
+  // If not present, it means the data service is not available to serve anything yet.
+  optional uint64 min_servable_version = 4;
+}
+
+message HistoricalDataServiceInfo {
+  optional aptos.util.timestamp.Timestamp timestamp = 1;
+  optional uint64 known_latest_version = 2;
+  optional StreamInfo stream_info = 3;
+}
+
+message FullnodeInfo {
```

Comment:
Do you think we should include validation here? like chain id?

Reply:
Good point. Done.